utils/components.py

```diff
@@ -61,3 +69,311 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+    configuration.ssl_ca_cert = f"{sa_path}/ca.crt"
+
+    model_endpoint = ""
+    model_name = ""
+    model_api_key = ""
+    namespace = open(f"{sa_path}/namespace").readline()
+
+    with client.ApiClient(configuration) as api_client:
+        core_api = client.CoreV1Api(api_client)
+
+        try:
+            config_map = core_api.read_namespaced_config_map(cm_name, namespace)
+            print(f"Reading configmap {cm_name} data...")
+            model_endpoint = config_map.data["endpoint"]
+            model_name = config_map.data["model"]
+        except ApiException as e:
+            print(f"""
+            ############################################ ERROR ###########################################################
+            # Configmap {cm_name} does not exist. Ensure you created a configmap with this name in namespace {namespace} #
+            ##############################################################################################################
+            """)
+            sys.exit(1)
+
+        try:
+            secret = core_api.read_namespaced_secret(secret_name, namespace)
+            print(f"Reading secret {secret_name} data...")
+            model_api_key = base64.b64decode(secret.data["api_key"]).decode("utf-8")
+        except ApiException as e:
+            print(f"""
+            ############################################## ERROR #######################################################
+            # Secret {secret_name} does not exist. Ensure you created a secret with this name in namespace {namespace} #
+            ############################################################################################################
+            """)
+            sys.exit(1)
+
+    request_auth = {"Authorization": f"Bearer {model_api_key}"}
+    request_body = {
+        "model": model_name,
+        "messages": [{"role": "user", "content": "tell me a funny joke."}],
+    }
+    resp = requests.post(
+        f"{model_endpoint}/chat/completions",
+        headers=request_auth,
+        data=json.dumps(request_body),
+        verify=os.environ["SDG_CA_CERT_PATH"],
+    )
+    if resp.status_code != 200:
+        print(f"""
+        ############################################ ERROR ####################################################
+        # Model Server {model_name} is unavailable. Ensure the model is up and it is ready to serve requests. #
+        #######################################################################################################
+        """)
+        sys.exit(1)
+    else:
+        print(f"""
+        ################### INFO #######################
+        # Model Server {model_name} is up and running. #
+        ################################################
+        """)
+
+
+@dsl.component(
+    base_image=PYTHON_IMAGE,
+    packages_to_install=["model-registry==0.2.13", "kserve==0.13"],
+)
+def test_model_registry(model_registry_endpoint: str):
```

Comment:
We will need to make this optional if `output_model_registry_name` (a pipeline input in the main branch now) is not specified.